### PR TITLE
AutoBrewer fixes

### LINKED ISSUE #3093: Auto brewer disappearing
### ❗ Checklist

- [X] I am using the official english version of Slimefun and did not modify the jar.
- [X] I am using an up to date "DEV" (not "RC") version of Slimefun.
- [X] I am aware that issues related to Slimefun addons need to be reported on their bug trackers and not here.
- [X] I searched for similar open issues and could not find an existing bug report on this.

### 📍 Description

When putting in a potion of fire resistance(3:00) and glowstone dust into the auto brewer, the auto brewer disappears after a few seconds.

### 📑 Reproduction Steps

1. place auto brewer 
2. hook auto brewer to energy
3. place fire resistance potion (3:00) into slot one 
4. place one glowstone dust into slot 2
5. close gui and wait a few seconds 

### 💡 Expected Behavior

Placing a potion into the auto brewer with  glowstone is supposed to strengthen the potion, for example if you put a swiftness potion in with glowstone it would make it a speed 2 potion, however fire resistance does not have a stronger potion type so the auto brewer is unable to actually auto brew the stronger potion.

### 📷 Screenshots / Videos

https://youtu.be/Pego5F9QNOQ

### 📜 Server Log

_No response_

### 📂 `/error-reports/` folder

_No response_

### 💻 Server Software

Purpur

### 🎮 Minecraft Version

1.16.x

### ⭐ Slimefun version

https://i.imgur.com/IAW5Nsk.png
https://i.imgur.com/6ur2Hda.png

### 🧭 Other plugins

_No response_

### LINKED ISSUE #3095: Fermenting a potion in the AutoBrewer removes redstone/glowstone modifiers
### ❗ Checklist

- [X] I am using the official english version of Slimefun and did not modify the jar.
- [X] I am using an up to date "DEV" (not "RC") version of Slimefun.
- [X] I am aware that issues related to Slimefun addons need to be reported on their bug trackers and not here.
- [X] I searched for similar open issues and could not find an existing bug report on this.

### 📍 Description

When trying to ferment instant health II potions in the AutoBrewer I found out that the output potion is a basic one, not the upgraded one.

### 📑 Reproduction Steps

1. Insert Potion of Instant Health II (or extended) and a fermented spider eye into the AutoBrewer
2. Wait for the process to finish
3. The output is a Potion of Harming I (or not extended)

### 💡 Expected Behavior

The output potion keeps its modifier (II or extended)

### 📜 Server Log

no error

### 📂 `/error-reports/` folder

no error

### 💻 Server Software

Paper

### 🎮 Minecraft Version

1.16.x

### ⭐ Slimefun version

DEV 927

### 🧭 Other plugins

none

### PR DESCRIPTION
**Pull request by martinbrom**
## Description

See attached issues

## Proposed changes

Check whether potions can be upgraded/extended before actually doing so.
When fermenting, pass the old potion values instead of false.

## Related Issues (if applicable)

Resolves #3093
Resolves #3095

## Checklist

- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [ ] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [x] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added.
- [x] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [ ] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/AutoBrewer.java [hunk 1]
```diff
@@ -52,10 +54,12 @@
...
     }
 
+    @ParametersAreNonnullByDefault
     public AutoBrewer(Category category, SlimefunItemStack item, RecipeType recipeType, ItemStack[] recipe) {
         super(category, item, recipeType, recipe);
     }
 
+    @Nullable
     @Override
     protected MachineRecipe findNextRecipe(BlockMenu menu) {
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/AutoBrewer.java:64]:
```suggestion
    @Override
    protected @Nullable MachineRecipe findNextRecipe(BlockMenu menu) {
```

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/AutoBrewer.java [hunk 2]
```diff
@@ -98,10 +102,13 @@ protected MachineRecipe findNextRecipe(BlockMenu menu) {
         }
     }
 
+    @Nullable
+    @ParametersAreNonnullByDefault
     private ItemStack brew(Material input, Material potionType, PotionMeta potion) {
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/AutoBrewer.java:107]:
```suggestion
    @ParametersAreNonnullByDefault
    private @Nullable ItemStack brew(Material input, Material potionType, PotionMeta potion) {
```

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/AutoBrewer.java [hunk 3]
```diff
@@ -150,11 +157,13 @@ private boolean isPotion(@Nonnull Material mat) {
...
 
+    @Nonnull
     @Override
     public ItemStack getProgressBar() {
         return new ItemStack(Material.FISHING_ROD);
     }
 
+    @Nonnull
     @Override
     public String getMachineIdentifier() {
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/AutoBrewer.java:168]:
```suggestion
    @Override
    public @Nonnull ItemStack getProgressBar() {
        return new ItemStack(Material.FISHING_ROD);
    }

    @Override
    public @Nonnull String getMachineIdentifier() {
```

**Review by TheBusyBiscuit — APPROVED** [review]
Minor styling remarks, otherwise looking good to me

**Review by TheBusyBiscuit — APPROVED** [review]

**Review by Sfiguz7 — APPROVED** [review]

**Pull request merged**